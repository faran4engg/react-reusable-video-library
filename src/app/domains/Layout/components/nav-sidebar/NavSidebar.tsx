import { FC } from 'react';
import Sidebar from './Sidebar';
import SidebarOverlay from './SidebarOverlay';
import { NavSidebarProps } from './types';

const NavSidebar: FC<NavSidebarProps> = ({
  isSidebarOpen,
  handleIsSidebarOpen,
}) => (
  <>
    <SidebarOverlay
      isSidebarOpen={isSidebarOpen}
      handleIsSidebarOpen={handleIsSidebarOpen}
    />
    <Sidebar
      isSidebarOpen={isSidebarOpen}
      handleIsSidebarOpen={handleIsSidebarOpen}
    />
  </>
);

export default NavSidebar;
