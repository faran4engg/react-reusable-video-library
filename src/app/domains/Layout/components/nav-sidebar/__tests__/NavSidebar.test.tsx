import React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';

import NavSidebar from '../NavSidebar';

const renderer = createRenderer();

describe('<NavSidebar />', () => {
  it('should render and match the snapshot', () => {
    renderer.render(
      <NavSidebar isSidebarOpen={false} handleIsSidebarOpen={jest.fn()} />,
    );
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
