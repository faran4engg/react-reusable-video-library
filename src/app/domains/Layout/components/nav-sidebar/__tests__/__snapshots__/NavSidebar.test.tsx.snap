// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavSidebar /> should render and match the snapshot 1`] = `
<React.Fragment>
  <SidebarOverlay
    handleIsSidebarOpen={[MockFunction]}
    isSidebarOpen={false}
  />
  <Sidebar
    handleIsSidebarOpen={[MockFunction]}
    isSidebarOpen={false}
  />
</React.Fragment>
`;
